src/cloud-cli/meltano/cloud/cli/config.py

```diff
@@ -119,11 +217,409 @@ async def set_value(
+            value_proc=VALIDATE_FUNCTION_DICT[notification_type],
+        )
+    )
+    VALIDATE_FUNCTION_DICT[notification_type](
+        value=input_value,
+        prompt_message=prompt_message,
+        error_message=error_message,
+    )
+    return input_value
+
+
+class FilterList(click.Option):
+    """Class for filters list for notifications."""
+
+    def type_cast_value(self, ctx: click.Context, value: list) -> list:
+        """Attempt to cast and validate filters parameter.
+
+        Args:
+            ctx: The click context
+            value: Multiple filters list gathered via --filter flag
+        """
+        filters = []
+        for item in value:
+            try:
+                filter_obj = json.loads(item)
+                filter_list = (
+                    filter_obj if isinstance(filter_obj, list) else [
+                        filter_obj]
+                )
+                for filter_dict in filter_list:
+                    validate_filter_dict(filter_dict)
+                filters.extend(filter_list)
+            except json.JSONDecodeError:
+                raise click.BadParameter(f"Unable to parse given json: {item}")
+
+        return filters
+
+
+def validate_filter_dict(filter_dict: dict) -> None:
+    """Helper function to validate filters dictionary against supported values.
+
+    Args:
+        filter_dict: The filter being validated
+    """
+    # For list of supported events check meltano/infra/common/models/meltano_models/models.py NotificationEventTypes
+    NOTIFICATION_EVENTS_SET = {"all"}
+    # For list of supported events check meltano/infra/common/models/meltano_models/models.py NotificationStatusfilters
+    NOTIFICATION_STATUS_FILTERS_SET = {"failed", "cancelled", "succeeded"}
+
+    if "events" in filter_dict:
+        # Since set() is a group for notificaiton need to set comprehension
+        events_set = {event for event in filter_dict['events']}
+        if not events_set.issubset(NOTIFICATION_EVENTS_SET):
+            raise click.BadParameter(
+                f"Some events not supported. Supported events: {NOTIFICATION_EVENTS_SET}, events: {filter_dict['events']}",
+            )
+
+    if "status" in filter_dict:
+        status_set = { status for status in filter_dict["status"]}
+        if not status_set.issubset(NOTIFICATION_STATUS_FILTERS_SET):
+            raise click.BadParameter(
+                f"Some status not supported. Supported status: {NOTIFICATION_STATUS_FILTERS_SET}, status: {filter_dict['status']}",
+            )
+
+
+@dataclass
+class WebhookNotificaton:
+    """Dataclass for webhook notifications."""
+
+    type: str
+    filters: list
+    webhook_url: str
+
+
+@dataclass
+class EmailNotification:
+    """Dataclass for email notifications."""
+
+    type: str
+    filters: list
+    email: str
+
+
+@config.group()
+def notification() -> None:
+    """Configure Meltano Cloud notifications."""
+
+
+@notification.group()
+def set() -> None:
```

Comment:
Having this set as a notification group command ends up breaking pythons built in set(). I tried to follow similar semantics from 
https://docs.meltano.com/reference/command-line-interface/#config

2 options is
1. Do similar and use set comprehension
2. Import set as a new name like pyset
3. Rename this command to put or something else

Curious about thoughts on this

Reply:
@leoroese You can rename the function to whatever you want, and then specify the name it should have in the CLI as a param to the `group` function